babel/support.py

```diff
@@ -175,11 +175,17 @@ def __init__(self, func, *args, **kwargs):
         object.__setattr__(self, '_kwargs', kwargs)
         object.__setattr__(self, '_is_cache_enabled', is_cache_enabled)
         object.__setattr__(self, '_value', None)
+        object.__setattr__(self, '_attribute_error', None)
 
     @property
     def value(self):
         if self._value is None:
-            value = self._func(*self._args, **self._kwargs)
+            try:
+                value = self._func(*self._args, **self._kwargs)
+            except AttributeError as error:
+                object.__setattr__(self, '_attribute_error', error)
+                raise AttributeError()
```

Comment:
One more nitpick – why construct a new error? You could just `raise` to raise the same `AttributeError` just stored.
```suggestion
                raise
```